### ISSUE TITLE
google_gke_hub_feature cluster upgrade upstream_fleets not working with project ID

### ISSUE THREAD

**Issue by benoitpuertolas** [hashicorp/terraform-provider-google#19350]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team) to expedite investigation and resolution of this issue.

### Terraform Version & Provider Version(s)

Terraform v1.6.5
on AMD64
+ provider registry.terraform.io/hashicorp/google v5.43.1
+ provider registry.terraform.io/hashicorp/google-beta v5.43.1

### Affected Resource(s)

google_gke_hub_feature

### Terraform Configuration

```tf
resource "google_gke_hub_feature" "cluster_upgrade" {
  name     = "clusterupgrade"
  location = "global"
  spec {
    clusterupgrade {
      upstream_fleets = [local.staging_project_id]  # i.e. infra-staging-xxxxxx
      post_conditions {
        soaking = "604800s"
      }
    }
  }
  project = local.project_id
}
```

### Expected Behavior

When doing running a second plan after creating the resource, no difference should be seen. 

### Actual Behavior

Plan:

```
  # google_gke_hub_feature.cluster_upgrade will be updated in-place
  ~ resource "google_gke_hub_feature" "cluster_upgrade" {
        id               = "projects/infra-tools-xxxxxxxx/locations/global/features/clusterupgrade"
        name             = "clusterupgrade"
        # (9 unchanged attributes hidden)

      ~ spec {
          ~ clusterupgrade {
              ~ upstream_fleets = [
                  - "xxxxxxxxxxxxx", # <-- Project Number
                  + "infra-staging-xxxxxxxx",  <-- Project ID
                ]

                # (1 unchanged block hidden)
            }
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

Apply:
```
Error: Error updating Feature "projects/infra-tools-xxxxxxxxx/locations/global/features/clusterupgrade": 
googleapi: Error 400: InvalidValueError for field upstream_fleets: upstream fleet ("infra-staging-xxxxxx") already has a downstream, cannot add a new one
```

The [documentation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/gke_hub_feature#upstream_fleets) says:

> The fleet name should be either fleet project number or id.
But actually only project number work properly (because it's the one saved in the state).

### Steps to reproduce

1. `terraform plan`
2. `terraform apply`
3. `terraform plan`
4. `terraform apply`

**Comment by shuyama1** [comment]
Yes, I can reproduce the issue in https://github.com/GoogleCloudPlatform/magic-modules/pull/11670. 

**Note to the service team:**

More details on errors and debug logs, please see https://github.com/GoogleCloudPlatform/magic-modules/pull/11670#issuecomment-2342084230
This [permadiff](https://googlecloudplatform.github.io/magic-modules/develop/permadiff/) s likely caused by the API returning project numbers, even when project IDs are explicitly specified in `spec.clusterupgrade.upstream_fleets`.
